pilot/utils/utils.py

```diff
@@ -34,23 +34,38 @@ def capitalize_first_word_with_underscores(s):
     return capitalized_string
 
 
-def get_prompt(prompt_name, original_data=None):
+def get_prompt(prompt_name, model=None, original_data=None):
     data = copy.deepcopy(original_data) if original_data is not None else {}
-
-    get_prompt_components(data)
+    
+    get_prompt_components(data, model)
 
     logger.info(f"Getting prompt for {prompt_name}")
 
-    # Load the template
-    template = env.get_template(prompt_name)
+    template = resolve_template(prompt_name, model)
 
     # Render the template with the provided data
     output = template.render(data)
 
     return output
 
+def resolve_template(prompt_name, model=None):
+    logger.debug(f'resolving prompt: {prompt_name}')
+
+    if model is not None:
+        model_prompt_name = f'{model}/{prompt_name}'
+        try:
+            # Attempt to return the model-specific template
+            return env.get_template(model_prompt_name)
+        except TemplateNotFound:
+            # If the model-specific template is not found, log the event or handle accordingly
+            logger.debug(f'Model-specific template not found for {model_prompt_name}. Falling back to general template.')
```

Comment:
While this is useful for testing, it would create too much noise if outputted for each model-specific template that's not overridden. I'd prefer to have just `return env.get_template(prompt_name)` here in the except clause.

Reply:
this is why I chose the debug level to log it out, maybe this should be resolve by setting he default level to info or something.